### PR TITLE
WFLY-13012 Update clustering integration tests to use Infinispan Serv…

### PR DESCRIPTION
**Pull request by rhusar**
…er 10.1.x

Resolves
https://issues.redhat.com/browse/WFLY-13012

### CODE REVIEW

**Comment by rhusar** [comment]
retest this please

**Comment by rhusar** [comment]
retest this please

File: testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/sso/remote/AbstractRemoteSingleSignOnTestCase.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import org.junit.rules.TestRule;
+
+/**
+ * @author Radoslav Husar
+ */
+public abstract class AbstractRemoteSingleSignOnTestCase extends AbstractSingleSignOnTestCase {
+
+    // Disable on Windows until https://issues.redhat.com/browse/ISPN-12041 is fixed
+    @ClassRule
+    public static final TestRule INFINISPAN_SERVER_RULE = Util.checkForWindows() ? new SkipJunit(OS.WINDOWS) : InfinispanServerRuleBuilder
```
Comment (pferraro) [inline testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/sso/remote/AbstractRemoteSingleSignOnTestCase.java:42]: Can we move INFINISPAN_SERVER_RULE to an interface?  That way we don't need to duplicate for every remote-cache-based test class hierarchy.

Comment (rhusar) [reply testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/sso/remote/AbstractRemoteSingleSignOnTestCase.java]: It appears it's not possible, it needs to be a static field on the test class. Nevertheless, I extracted the `org.junit.rules.TestRule` into a separate method, so the setup is just 1 annotated field.

**Review by pferraro — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks, @rhusar

**Pull request merged**